src/mozanalysis/stats/summarize_samples.py

```diff
@@ -0,0 +1,195 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import numpy as np
+import pandas as pd
+
+
+default_quantiles = (0.005, 0.05, 0.5, 0.95, 0.995)
+
+
+def summarize_one_branch_samples(samples, quantiles=default_quantiles):
+    """Return descriptive statistics for sampled population-level stats.
+
+    Given samples from a distribution, calculate some quantiles and the
+    mean.
+
+    The intended primary use-case is for calculating confidence
+    intervals when bootstrapping, or probability bands around model
+    parameters; in both cases `samples` relate to data from one branch
+    of an experiment.
+
+    Args:
+        samples (pandas.Series): Samples over which to compute the mean
+            and quantiles.
+        quantiles (list, optional): The quantiles to compute - a good
+            reason to override the defaults would be when Bonferroni
+            corrections are required.
+
+    Returns a pandas Series; the index contains the stringified
+    `quantiles` plus `'mean'`.
+    """
+    if not isinstance(samples, (pd.Series, np.ndarray, list)):
+        # Hey pd.Series.agg - don't apply me elementwise!
+        # Raising this error allows `summarize_one_branch_samples_batch`
+        # to work also for non-batch `samples` (i.e. doing double duty)
+        raise TypeError("Can't summarize a scalar")
+
+    q_index = [str(v) for v in quantiles]
+
+    res = pd.Series(index=q_index + ['mean'])
+
+    res[q_index] = np.quantile(samples, quantiles)
```

Comment:
Thanks, glad to add some nuance to my bootstrapping. I have a few questions to verify that I'm on the right track.

Is it correct to say that the flaws in the percentile bootstrap arise from errors in the approximation of the true cumulative distribution function by the empirical distribution function? So that when we sample from Bayesian models, the functions in `summarize_samples.py` are valid in their current state, and this correction is only relevant when bootstrapping? In that case I'll put the corrections over in `bootstrap.py`.

In a world where we're not willing to make the approximation that a statistic over a bootstrapped sample is a statistic over another sample from the true distribution, I bet that `summarize_joint_samples` needs to be a lot more complicated... Is there a canonical approach to that problem, or do we just have to hope that each branch's behaviour is such that our samples for each branch are biased in a similar way and they cancel out when we compare them?

Reply:
I claim to have fixed this (for the one-branch case) with my most recent commit.

Reply:
The bootstrapping routine in the frequentist module will now be to your liking w.r.t. this comment.